### PR TITLE
Add docker support and add docker-chs-dev to README

### PR DESCRIPTION
**Pull request by amoriartyCH**
Dockerignore file added and updated 
Dockerfile created
Go modules updated

README updated to include new information.

**Related PR in docker-chs-development**

https://github.com/companieshouse/docker-chs-development/pull/227

https://github.com/companieshouse/chs-delta-api/pull/3

**Resolves**:
DSND-26 [https://companieshouse.atlassian.net/browse/DSND-26]

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,10 +1,10 @@
 module github.com/companieshouse/chs-delta-api
 
-go 1.12
+go 1.15
```
Comment (divya051988) [inline go.mod:3]: Are we sure we want to use go 1.15?? This will need creation of new. tasks in concourse and probably we need to let platform know?

**Comment by divya051988** [comment]
Can you please add a resolves section to the Description - https://github.com/companieshouse/styleguides/blob/master/git/pull-requests.md

File: go.mod [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
 
 require (
-	github.com/companieshouse/chs.go v1.2.1
+	github.com/companieshouse/chs.go v1.2.4
 	github.com/companieshouse/gofigure v0.1.4
 	github.com/gorilla/mux v1.7.3
```
Comment (divya051988) [inline go.mod:8]: Can we use 1.8.0 for gorilla/mux ?

**Comment by divya051988** [comment]
Also might help to add a link to the PR in chs-docker-development repo to get a visual of the modules mentioned in the README

File: go.mod (hunk 1 above)
Comment (amoriartyCH) [reply go.mod:3]: For transparency. We will be going with go v1.12 and on the side we will get support and begin implementing latest versions in concourse etc (v1.15+...).

File: go.mod (hunk 2 above)
Comment (amoriartyCH) [reply go.mod:8]: I will update this. Also to address the comment below this one, I will add to the description of this PR a link to the docker-chs-development PR when it is created.

**Review by divya051988 — APPROVED** [review]

**Pull request merged**